File: kotlin/src/test/kotlin/PeopleCounterTest.kt
package com.vo

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Disabled
import org.junit.jupiter.api.Test

class PeopleCounterTest {

    @Test
    fun `should return correct occupancy`() {
        val httpClient = HttpClient()
        val cameras = Cameras(listOf(
            Camera(httpClient, "Camera1", "192.168.55.130"),
            Camera(httpClient, "Camera2", "192.168.55.131"),
            Camera(httpClient, "Camera3", "192.168.55.132"),
            Camera(httpClient, "Camera4", "192.168.55.133"),
            Camera(httpClient, "Camera5", "192.168.55.134")
        ))
        val lobby = Zone(
            "South Auditorium", cameras
        )

        lobby.update()
        assertEquals(lobby.occupancy, 25)
    }

    @Disabled
    @Test
    fun `should handle errors`() {}
}

File: csharp/people-counter/Domain.Tests/PeopleCounterTest.cs
using System.Collections.Generic;
using Domain;
using Xunit;

namespace Domain.Tests
{
    public class PeopleCounterTest
    {
        [Fact]
        public void ShouldReturnCorrectOccupancy()
        {
            var httpClient = new HttpClient();
            var lobby = new Zone(
                "South Auditorium", new List<Camera> {
                    new Camera("Camera1", "192.168.55.130"),
                    new Camera("Camera2", "192.168.55.131"),
                    new Camera("Camera3", "192.168.55.132"),
                    new Camera("Camera4", "192.168.55.133"),
                    new Camera("Camera5", "192.168.55.134")
                }
            );

            var cameras = new Cameras(httpClient);
            var occupancy = cameras.GetFor(lobby);

            Assert.Equal(25, occupancy.Total);
        }
    }
}
